src/lexer/init.lua

```diff
@@ -263,6 +264,51 @@ function lexer.navigator()
 		end)
 	end
 
+	function nav:HotswapSource(newSource)
+		local oldSource = self.Source or ""
```

Comment:
If the old source is nil or empty, we don't have to do all this work, right? There's no tokens to keep.
If the new source is nil or empty the same concept applies, as there's no old tokens needed.

Let's skip all the diffing work to just call `SetSource` and return early for some common cases.

```Luau
if (#oldSource == 0) or (#newSource == 0) or (string.byte(oldSource, 1) ~= string.byte(newSource, 1)) then
	-- No common tokens to share, just set normally
	self:SetSource(newSource)
	return
end
```